extensions/azure-keyvault/runtime/src/main/java/io/quarkiverse/azure/keyvault/secret/runtime/config/KeyVaultSecretConfigUtil.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkiverse.azure.keyvault.secret.runtime.config;
+
+import com.azure.security.keyvault.secrets.models.KeyVaultSecretIdentifier;
+
+public class KeyVaultSecretConfigUtil {
+    private static final String AZURE_KEYVAULT_PREFIX = "kv//";
+    private static final String AZURE_KEYVAULT_ENDPOINT_PREFIX = "https://";
+    private static final String AZURE_VAULT_URL_FORMAT = "https://%s.vault.azure.net/secrets/%s/%s";
+
+    private KeyVaultSecretConfigUtil() {
+    }
+
+    static KeyVaultSecretIdentifier getSecretIdentifier(String input, String defaultEndpoint) {
+
+        if (!input.startsWith(AZURE_KEYVAULT_PREFIX)) {
+            return null;
+        }
+
+        String resourcePath = input.substring(AZURE_KEYVAULT_PREFIX.length());
+        String[] tokens = resourcePath.split("/");
+
+        String kvName = null;
+        String secretName = null;
+        String version = "latest";
+
+        if (tokens.length == 1) {
+            // property is form "kv//<secret-name>"
+            kvName = getAzureKeyVaultName(defaultEndpoint);
+            secretName = tokens[0];
+        } else if (tokens.length == 2) {
+            // property is form "kv//<kv-name>/<secret-name>"
+            kvName = tokens[0];
+            secretName = tokens[1];
+        } else if (tokens.length == 3
+                && tokens[1].equals("secrets")) {
+            // property is form "kv//<kv-name>/secrets/<secret-name>"
+            kvName = tokens[0];
+            secretName = tokens[2];
+        } else if (tokens.length == 3
+                && tokens[1].equals("versions")) {
+            // property is form "kv//<secret-name>}/versions/<version>"
+            kvName = getAzureKeyVaultName(defaultEndpoint);
+            secretName = tokens[0];
+            version = tokens[2];
+        } else if (tokens.length == 4
+                && tokens[1].equals("secrets")) {
+            // property is form "kv//<kv-name>/secrets/<secret-name>/<version>"
+            kvName = tokens[0];
+            secretName = tokens[2];
+            version = tokens[3];
+        } else if (tokens.length == 5
+                && tokens[1].equals("secrets")
+                && tokens[3].equals("versions")) {
+            // property is form "kv//<kv-name>/secrets/<secret-name>/versions/<version>"
+            kvName = tokens[0];
+            secretName = tokens[2];
+            version = tokens[4];
+        } else {
```

Comment:
Such `if/else` blocks give me a bad code smell. I understand this is a straightforward way to implement the specified syntax here, but are you sure you captured all the corner cases? Is this rigorously tested?

Reply:
I discussed this with @galiacheng. The positions are completely unambiguous. So the if/else is ok.